pkg/job/runtime_v2/kubernetes_runtime.go

```diff
@@ -508,6 +503,148 @@ func (kr *KubeRuntime) GetJobLog(jobLogRequest pfschema.JobLogRequest) (pfschema
 	return jobLogInfo, nil
 }
 
+// GetEvents get events by name and namespace
+func (kr *KubeRuntime) GetEvents(namespace, name string) ([]corev1.Event, error) {
+	log.Infof("get %s/%s events info from Kubernetes", namespace, name)
+	var response []corev1.Event
+	listOptions := metav1.ListOptions{
+		FieldSelector: fmt.Sprintf("involvedObject.name=%s", name),
+	}
+	// current k8s unsupported 'sort by time'
+	events, err := kr.clientset().CoreV1().Events(namespace).List(context.TODO(), listOptions)
+	if err != nil {
+		log.Errorf("list events for resource %s/%s failed, err: %v", namespace, name, err)
+		return response, err
+	}
+	return events.Items, nil
+}
+
+// GetPod get pod by namespace and name
+func (kr *KubeRuntime) GetPod(namespace, name string) (corev1.Pod, error) {
+	log.Debugf("get kubernetes pod: %s/%s", namespace, name)
+	pod, err := kr.clientset().CoreV1().Pods(namespace).Get(context.TODO(), name, metav1.GetOptions{})
```

Comment:
It might be better to use `kr.clientset().CoreV1(),Pods(string).Get(x, x, x)`

Reply:
ok